### DISCUSSION TITLE
Location spaces

### DISCUSSION THREAD

**Discussion by Thatoo** [element-hq/element-meta#485] [Ideas]
#### What would you like to do?
The user could be able to browse community space members by location around herself/himself.
#### Why would you like to do it?
Being able to find people living around (or staying for a while in the neighborhood) who belong to the same community (passion like astronomy, political subject, religion, LGBTQ+...) as the user.
#### How would you like to achieve it?
As there is video room which are specific room dedicated to video chat, it would be possible to have location space, a specific type of space where users who are member of accept to share their location in order to be able to be contacted by other users nearby and be able to see a list of users nearby and contact them. Of course the space would have a subject, a community driven aspect.
The user should be able to set the number of km around he accepts to be visible by others.

**Comment by Thatoo** [comment]
Well, it doesn't need to be a specific space actually, It might not be a space setting after all.
It should be a user choice in each space!
I feel that it's probably easier to implement and it gives the choice to the user.
When a user join a space, as it can set notifications it could set if she/he want's to share his location to other nearby users (how many X km around) of this specific space and be able to be contacted by them.
It's not sharing the exact location of course, that would be too private to share to yet unknown people, but be highlighted in a way to nearby users : be able to sort the list of users by distance nearby the user, only nearby users who have specifically accepted to appear in the list of users being around would be listed.

**Comment by Thatoo** [comment]
Sorry, I try to clarify my thoughts... and make an example :
We have a Space called AAA related to a specific subject
We have X users in this space : user 1, user 2...
User 1 set that he accepts to be visible by people around him in a 10 km radius circle.
User 2 lives at 5 km from User 1 but don't want to be visible, so he won't be visible by user 1 and he won't be able to sort users by distance
User 3 lives also at 5 km from User 1 and set he accepts to be visible by people around him in a 15 km radius circle
User 1 and 3 will be able to see each other if they sort the list of users by distance. User 3 would not see User 2 neither.
User 4 lives at 15 km from User 1 and 10 km form User 3 and he accepts to be visible by people around him in a 15 km radius circle.
User 3 and 4 would be able to see each other if they sort the list of users by distance. User 4 would not be able to see User 1 because he is out of the 10 km radius accepted by User 1 but he would be visible by User 1 because he accepted to be visible by people living at 15 km and User 1 would see that User 4 is at 15 km from him. It should be clear in the UI design for User 1 that he can see User 4 because User 4 accepted a circle of 15 km but that User 4 can't see him.

Of course the default behavior when a user join a space should be to hide the location. The user should be really aware when he decide to be visible by nearby people.

I hope I manage to be clearer with this example.

**Comment by Thatoo** [comment]
Related to https://github.com/vector-im/element-meta/discussions/1341

**Comment by t3chguy** [comment]
Closing in favour of https://github.com/vector-im/element-meta/discussions/1341
